client/relay-chain-minimal-node/src/network.rs

```diff
@@ -0,0 +1,420 @@
+		role: config.role.clone(),
+		executor: {
+			let spawn_handle = Clone::clone(&spawn_handle);
+			Some(Box::new(move |fut| {
+				spawn_handle.spawn("libp2p-node", Some("networking"), fut);
+			}))
+		},
+		transactions_handler_executor: {
+			let spawn_handle = Clone::clone(&spawn_handle);
+			Box::new(move |fut| {
+				spawn_handle.spawn("network-transactions-handler", Some("networking"), fut);
+			})
+		},
+		fork_id: None,
+		chain_sync: Box::new(DummyChainSync),
+		network_config: config.network.clone(),
+		chain: client.clone(),
+		transaction_pool: transaction_pool_adapter as _,
+		import_queue: Box::new(DummyImportQueue),
+		protocol_id,
+		metrics_registry: config.prometheus_config.as_ref().map(|config| config.registry.clone()),
+		block_request_protocol_config,
+		state_request_protocol_config,
+		warp_sync_protocol_config: None,
+		light_client_request_protocol_config,
+		request_response_protocol_configs: Vec::new(),
+	};
+
+	let network_mut = sc_network::NetworkWorker::new(network_params)?;
+	let network = network_mut.service().clone();
+
+	let future = build_network_collator_future(network_mut);
+
+	let (network_start_tx, network_start_rx) = futures::channel::oneshot::channel();
+
+	// The network worker is responsible for gathering all network messages and processing
+	// them. This is quite a heavy task, and at the time of the writing of this comment it
+	// frequently happens that this future takes several seconds or in some situations
+	// even more than a minute until it has processed its entire queue. This is clearly an
+	// issue, and ideally we would like to fix the network future to take as little time as
+	// possible, but we also take the extra harm-prevention measure to execute the networking
+	// future using `spawn_blocking`.
+	spawn_handle.spawn_blocking("network-worker", Some("networking"), async move {
+		if network_start_rx.await.is_err() {
+			tracing::warn!(
+				"The NetworkStart returned as part of `build_network` has been silently dropped"
+			);
+			// This `return` might seem unnecessary, but we don't want to make it look like
+			// everything is working as normal even though the user is clearly misusing the API.
+			return
+		}
+
+		future.await
+	});
+
+	let network_starter = NetworkStarter::new(network_start_tx);
+
+	Ok((network, network_starter))
+}
+
+/// Builds a never-ending future that continuously polls the network.
+///
+/// The `status_sink` contain a list of senders to send a periodic network status to.
+async fn build_network_collator_future<B: BlockT, H: sc_network::ExHashT, C: HeaderBackend<B>>(
+	mut network: sc_network::NetworkWorker<B, H, C>,
+) {
+	loop {
+		futures::select! {
+			// Answer incoming RPC requests.
+
+			// The network worker has done something. Nothing special to do, but could be
+			// used in the future to perform actions in response of things that happened on
+			// the network.
+			_ = (&mut network).fuse() => {}
+		}
+	}
+}
```

Comment:
```suggestion
```

The `NetworkWorker` is already a future and a future "fires once" and then you should not poll it anymore.